docs/ocis/adr/0030-user-generated-client-keys.md

```diff
@@ -0,0 +1,93 @@
+---
+title: "30. User generated client keys"
+date: 2024-02-22T15:30:00+01:00
+weight: 30
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0030-user-generated-client-keys.md
+---
+
+* Status: draft
+* Deciders: [@butonic](https://github.com/butonic), [@fschade](https://github.com/fschade), [@kulmann](https://github.com/kulmann), [@dragotin](https://github.com/dragotin), [@rhafer](https://github.com/rhafer)
+* Date: 2024-02-22
+
+## Context and Problem Statement
+
+Currently, the oCIS proxy is used to authenticate users based on an OpenID Connect access token or using basic auth.
+The current implementation is limited to a single identity provider, and we currently have no way to generate credentials for cli tools that do not yet support OpenID Connect
+
+The coming up calendar and contacts service requires this because most CalDAV and CardDAV clients require basic or digest authentication.
+
+## Decision Drivers <!-- optional -->
+
+## Considered Options
+
+* oCIS proxy middleware with micro store
+
+## Decision Outcome
+
+Chosen option: *???*
+
+### Positive Consequences:
+
+* TODO
+
+### Negative Consequences:
+
+* TODO
+
+## Pros and Cons of the Options <!-- optional -->
+
+### oCIS proxy middleware with micro store
+
+We use the built-in micro store to save generated access keys.
+Generated access keys will have a scope attached to limit access to oCIS services or endpoints.
+
+These scopes need to be defined - for ccs it would be sufficient to use /ccs/ endpoint.
+
+The generated access keys need to be entered into classic authentication forms which consist of user and password.
+Both elements need to be provided/generated by oCIS to the user.
+
+There will be a web frontend where users can:
+- create a new pair of user and password(key) for selected scopes and give some display name.
+- list existing keys with display name, scope, user (but not the password - or shall we???)
```

Comment:
no ... will only lead to weak passwords. maybe allow the admin to set a custom password for migration purposes?